python/cudf/dataframe/dataframe.py

```diff
@@ -469,18 +469,22 @@ def take(self, positions, ignore_index=False):
             out[col] = self[col].take(positions, ignore_index=ignore_index)
         return out
 
-    def copy(self):
+    def copy(self, deep=True):
         """
         Returns a copy of this dataframe
         """
         df = DataFrame()
         df._index = self._index
```

Comment:
This is a pass by reference and needs to be properly deep copied if deep=True.

Reply:
Indeed, my oversight.

Reply:
This should result in an Index copy function as well and just call `self._index.copy()`